luminance-derive/src/semantics.rs

```diff
@@ -29,6 +30,21 @@ impl fmt::Display for SemanticsImplError {
   }
 }
 
+impl error::Error for SemanticsImplError {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+	match self {
+	    SemanticsImplError::AttributeErrors(ref ve) => {
+		if ve.len() > 0 {
+		    return Some(&ve[0])
+		} else {
+		    return None;
+		}
+	    },
+	    _ => None
```

Comment:
You can simplify this with:

```suggestion
	    SemanticsImplError::AttributeErrors(ref ve) => ve.first()
```